apis/goharbor.io/v1alpha2/chartmuseum_types.go

```diff
@@ -0,0 +1,356 @@
+package v1alpha2
+
+import (
+	harbormetav1 "github.com/goharbor/harbor-operator/apis/meta/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// +kubebuilder:object:root=true
+// +kubebuilder:storageversion
+// +k8s:openapi-gen=true
+// +resource:path=chartmuseum
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:categories="goharbor"
+// +kubebuilder:printcolumn:name="Version",type=string,JSONPath=`.spec.version`,description="The semver Harbor version",priority=5
+// +kubebuilder:printcolumn:name="Replicas",type=string,JSONPath=`.spec.replicas`,description="The number of replicas",priority=0
+// ChartMuseum is the Schema for the registries API.
+type ChartMuseum struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec ChartMuseumSpec `json:"spec,omitempty"`
+
+	Status harbormetav1.ComponentStatus `json:"status,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+// ChartMuseumList contains a list of ChartMuseum.
+type ChartMuseumList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []ChartMuseum `json:"items"`
+}
+
+// ChartMuseumSpec defines the desired state of ChartMuseum.
+type ChartMuseumSpec struct {
+	harbormetav1.ComponentSpec `json:",inline"`
+
+	// +kubebuilder:validation:Optional
+	Log ChartMuseumLogSpec `json:"log,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	Auth ChartMuseumAuthSpec `json:"auth,omitempty"`
```

Comment:
I think we don't need to expose auth related config. It's a fixed setting.

```
# Credential for internal communication
BASIC_AUTH_USER=chart_controller
BASIC_AUTH_PASS=
```

Reply:
I suggest t add comments to warn about that.
Following discussion on <https://github.com/goharbor/harbor-operator/pull/59#discussion_r467071357>